geode-core/src/main/java/org/apache/geode/net/SSLParameterExtension.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.net;
+
+import javax.net.ssl.SSLParameters;
+
+import org.apache.geode.distributed.internal.DistributionConfig;
+
+/**
+ * User implementation of a SSLParameter extension logic.
+ *
+ * @since Geode 1.12
+ */
+
+public interface SSLParameterExtension {
+  /**
+   * Initialize the SSLParameterExtension.
+   *
+   * @param config the DistributionConfig
+   */
+  default void init(DistributionConfig config) {}
+
+  SSLParameters modifySSLClientSocketParameters(SSLParameters parameters);
+
+  SSLParameters modifySSLServerSocketParameters(SSLParameters parameters);
```

Comment:
I think both of these should just have default implementations of `return parameters`. I would think that most implementations would only require one or the other of the methods to actually be implemented.

Reply:
Thanks for comments